src/ast/passes/semantic_analyser.cpp

```diff
@@ -3553,60 +3603,56 @@ SizedType SemanticAnalyser::create_key_type(const SizedType &expr_type,
   return new_key_type;
 }
 
-void SemanticAnalyser::update_key_type(const Map &map,
-                                       const SizedType &new_key_type)
+void SemanticAnalyser::maybe_update_current_key(SizedType &current_key_type,
```

Comment:
Nit: is the `maybe_` prefix necessary? It's sort of expected that the key won't be updated if not necessary.

I'd even go with `update_key` or `update_map_key`.

Reply:
Pretty bike-sheddy 🖌️ 🚲 🏠  but I would argue my function name (though more verbose) is the most accurate/descriptive.

Reply:
No problem, I'm certainly not going to fight for it :grinning: Feel free to keep it as it is.

Reply:
Ok fine, I'll get rid of the `maybe`.